containers/unit_tests/TestUnorderedMap.hpp

```diff
@@ -480,6 +480,48 @@ TEST(TEST_CATEGORY, UnorderedMap_consistent_size) {
   ASSERT_EQ(2u, m.size());
 }
 
+namespace Impl {
+
+using map_type = Kokkos::UnorderedMap<int, void, Kokkos::DefaultExecutionSpace>;
+
+KOKKOS_FUNCTION
+void test_insert_to_map_copy(map_type const &input_map, const int i) {
+  auto map = input_map;
+  map.insert(i);
+}
+
+struct TestMapCopy {
+  map_type m_map;
+
+  KOKKOS_FUNCTION
+  void operator()(const int i) const { test_insert_to_map_copy(m_map, i); }
+};
+
+}  // namespace Impl
+
+TEST(TEST_CATEGORY, UnorderedMap_shallow_copyable_on_device) {
+  Impl::TestMapCopy test_map_copy;
+
+  Kokkos::parallel_for(1, test_map_copy);
+  ASSERT_EQ(1u, test_map_copy.m_map.size());
+}
+
+#if !defined(KOKKOS_ENABLE_CUDA) || \
+    (defined(KOKKOS_ENABLE_CUDA) && defined(KOKKOS_ENABLE_CUDA_LAMBDA))
+void test_unordered_map_device_capture() {
```

Comment:
Would this test actually fail without this patch?

Reply:
Yes, without this patch, compiling this lambda capture test will produce multiple warnings/errors like this:
`calling a __host__ function("std::__shared_count<( ::__gnu_cxx::_Lock_policy)2> ::~__shared_count") from a __host__ __device__ function("Kokkos::UnorderedMap<int, void,  ::Kokkos::Cuda,  ::Kokkos::pod_hash<int> ,  ::Kokkos::pod_equal_to<int> > ::~UnorderedMap [subobject]") is not allowed`.

It is a little different compile error output from the shallow_copy test above, but they are both produced by having a `shared_ptr` in `UnorderedMap` and are resolved with this patch.